xbmc/music/infoscanner/MusicInfoScanner.cpp

```diff
@@ -112,7 +104,14 @@ void CMusicInfoScanner::Process()
         m_handle = dialog->GetHandle(g_localizeStrings.Get(314));
     }
 
-    m_bClean = g_advancedSettings.m_bMusicLibraryCleanOnUpdate;
+    // check if we only need to perform a cleaning
+    if (m_bClean && m_pathsToScan.empty())
+    {
+      CleanDatabase(false);
+      m_bRunning = false;
+
+      return;
```

Comment:
I'm happy to have things pointed out when I am being blind. :)

`CVideoInfoScanner` does similar with progress, but cleaning branch does set  `m_handle = NULL`
https://github.com/xbmc/xbmc/blob/efe979aec099a7fb2bb13ff235280ba1fdf55cb8/xbmc/video/VideoInfoScanner.cpp#L92

What do you think?